beacon-chain/rpc/prysm/v1alpha1/validator/proposer_deposits.go

```diff
@@ -107,9 +109,32 @@ func (vs *Server) deposits(
 	// deposits are sorted from lowest to highest.
 	var pendingDeps []*ethpb.DepositContainer
 	for _, dep := range allPendingContainers {
-		if uint64(dep.Index) >= beaconState.Eth1DepositIndex() && uint64(dep.Index) < canonicalEth1Data.DepositCount {
-			pendingDeps = append(pendingDeps, dep)
+		if beaconState.Version() < version.Electra {
+			// min(MAX_DEPOSITS, eth1_data.deposit_count - state.eth1_deposit_index)
```

Comment:
I dont understand this comment. What does it have to do with the next line?

Reply:
this was not that clear to me at first either, but the intension of the loop is to get the number of pending deposits it reads differently in the python spec from the actual implementation.

Reply:
https://github.com/ethereum/consensus-specs/blob/dev/specs/electra/validator.md you can see it here, the comment was just so show what the python specs represents previously vs newly with electra

Reply:
it's sort of broken up for us, the max deposits check is on line 139, the if statement appends pending deposits to a list which is representing this in the phase0 spec

```
If there are any unprocessed deposits for the existing state.eth1_data (i.e. state.eth1_data.deposit_count > state.eth1_deposit_index), then pending deposits must be added to the block. The expected number of deposits is exactly min(MAX_DEPOSITS, eth1_data.deposit_count - state.eth1_deposit_index). 
```

Reply:
OK, how about this?

```suggestion
			// Add deposits up to min(MAX_DEPOSITS, eth1_data.deposit_count - state.eth1_deposit_index)
```